File: app/src/main/java/com/example/reminders20/di/modules/ViewModelModule.java
package com.example.reminders20.di.modules;

import com.example.reminders20.db.Reminder;
import com.example.reminders20.db.ReminderDao;
import com.example.reminders20.viewModels.ListViewModel;
import com.example.reminders20.viewModels.NewReminderViewModel;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ViewModelModule {

    @Provides
    @Singleton
    ListViewModel provideListViewModel(ReminderDao reminderDao) {
        return new ListViewModel(reminderDao);
    }
    @Provides
    NewReminderViewModel provideNewReminderViewModel(ReminderDao reminderDao) {
        return new NewReminderViewModel(reminderDao);
    }

}

File: app/src/main/java/com/example/reminders20/di/modules/ViewModelModule.kt
package com.example.reminders20.di.modules

import com.example.reminders20.db.ReminderDao
import com.example.reminders20.viewModels.ListViewModel
import com.example.reminders20.viewModels.NewReminderViewModel
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class ViewModelModule {
    @Provides
    @Singleton
    fun provideListViewModel(reminderDao: ReminderDao): ListViewModel {
        return ListViewModel(reminderDao)
    }

    @Provides
    fun provideNewReminderViewModel(reminderDao: ReminderDao): NewReminderViewModel {
        return NewReminderViewModel(reminderDao)
    }
}